lib/charms/loki_k8s/v0/loki.py

```diff
@@ -115,22 +375,98 @@ def __init__(self, charm: CharmBase, relation_name: str):
                 `self` in the instantiating class.
             relation_ name: a string name of the relation between `charm` and
                 the Loki charmed service.
+            alert_rules_path: an optional path for the location of alert rules
+                files.  Defaults to "src/loki_alert_rules" at the top level
+                of the charm repository.
         """
         super().__init__(charm, relation_name)
         self._stored.set_default(loki_push_api=None)
         self._charm = charm
         self._relation_name = relation_name
+        self._ALERT_RULES_PATH = alert_rules_path
         events = self._charm.on[relation_name]
         self.framework.observe(events.relation_changed, self._on_logging_relaton_changed)
 
     def _on_logging_relaton_changed(self, event):
+        """Handle changes in related consumers.
+
+        Anytime there are changes in the relation between Loki
+        and its consumers charms,
+
+        Args:
+            event: a `CharmEvent` in response to which the consumer
+                charm must update its relation data.
+        """
         if event.unit is None:
             # Workaround: Seems this is a Juju bug that sends event.unit == None
             return
 
         if data := event.relation.data[event.unit].get("data"):
             self._stored.loki_push_api = json.loads(data)["loki_push_api"]
 
+        event.relation.data[self._charm.app]["metadata"] = json.dumps(self._scrape_metadata)
+        self._set_alert_rules(event)
+
+    def _set_alert_rules(self, event):
+        """Set alert rules into relation data.
+
+        Args:
+            event: a `CharmEvent` in response to which the consumer
+                charm must update its relation data.
+        """
+        event.relation.data[self._charm.unit]["loki_host"] = str(
```

Comment:
Could you please document (if haven't already) what goes into app data and what into unit data?

Reply:
This relation data is not really necessary. Removed.